allennlp/models/model.py

```diff
@@ -78,15 +78,18 @@ def __init__(self, vocab: Vocabulary, regularizer: RegularizerApplicator = None)
         self.vocab = vocab
         self._regularizer = regularizer
 
-    def get_regularization_penalty(self) -> Union[float, torch.Tensor]:
+    def get_regularization_penalty(self) -> Optional[torch.Tensor]:
         """
         Computes the regularization penalty for the model.
-        Returns 0 if the model was not configured to use regularization.
+        Returns None if the model was not configured to use regularization.
         """
         if self._regularizer is None:
-            return 0.0
+            regularization_penalty = None
         else:
-            return self._regularizer(self)
+            regularization_penalty = self._regularizer(self)
+            if isinstance(regularization_penalty, float):
+                regularization_penalty = torch.tensor(regularization_penalty)
```

Comment:
If we get a `float` from `self._regularizer`, how does that work with gradients? I don't see how regularization can work if we lose track of the computation graph here. `float` doesn't have any information about the computation graph.

I guess it works if the `float` is 0. Is that what this case is capturing?

Reply:
That's what I also thought. But this test case: `TestElmoTokenEmbedder.test_tagger_with_elmo_token_embedder_can_train_save_and_load` seems to require explicit conversion to tensor. The value is actually 0.0 .

Reply:
I see. This happens in the applicator when none of the regexes match. That's probably not a great situation, since it might mean the config doesn't do what you think it does. But let's not invent a whole new problem area to solve in this PR.

In this case, let's throw an exception when the number isn't `0.0` and call it done?